src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -398,6 +426,20 @@ public boolean updateUserGroup(UserGroup userGroup, List<Permission> permissions
 			groupPermissionIoOperationRepository.saveAll(groupPermissions);
 		}
 
-		return updated;
+		return findByCode(userGroup.getCode());
+	}
+
+	public void ensureUserNotDeleted(String username) throws OHServiceException {
+		User entity = repository.findByUserNameAndDeleted(username, true);
+		if (entity != null) {
+			throw new OHServiceException(new OHExceptionMessage("angal.common.denied.msg"));
+		}
+	}
+
+	public void ensureUserGroupNotDeleted(String code) throws OHServiceException {
+		UserGroup entity = groupRepository.findByCodeAndDeleted(code, true);
+		if (entity != null) {
+			throw new OHServiceException(new OHExceptionMessage("angal.common.denied.msg"));
+		}
```

Comment:
Same `MessageBundle.getMessage()` missing here and keys to be changed in case my proposal is accepted.